core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -466,700 +626,845 @@ private void buildChannel() {
-         relay2.setRouteStatusListener(new DefaultRouteStatusListener());
+   private CompletableFuture<Map<Address, Response>> performSyncRemoteInvocation(
+         Collection<Address> targets, ReplicableCommand command, ResponseMode mode, long timeout,
+         ResponseFilter responseFilter, DeliverOrder deliverOrder, boolean ignoreLeavers, boolean sendStaggeredRequest,
+         boolean broadcast, Address singleTarget) {
+      CompletableFuture<Map<Address, Response>> request;
+      if (sendStaggeredRequest) {
+         FilterMapResponseCollector collector = new FilterMapResponseCollector(responseFilter, false, targets.size());
+         request = invokeCommandStaggered(targets, command, collector, deliverOrder, timeout, TimeUnit.MILLISECONDS);
+      } else {
+         if (singleTarget != null) {
+            ResponseCollector<Map<Address, Response>> collector =
+                  ignoreLeavers ? SINGLE_IGNORE_LEAVERS_COLLECTOR : SINGLE_RESPONSE_COLLECTOR;
+            request = invokeCommand(singleTarget, command, collector, deliverOrder, timeout, TimeUnit.MILLISECONDS);
+         } else {
+            ResponseCollector<Map<Address, Response>> collector;
+            if (mode == ResponseMode.WAIT_FOR_VALID_RESPONSE) {
+               collector = new FilterMapResponseCollector(responseFilter, false, targets.size());
+            } else if (responseFilter != null) {
+               collector = new FilterMapResponseCollector(responseFilter, true, targets.size());
+            } else {
+               collector = new SyncMapResponseCollector(ignoreLeavers, targets.size());
+            }
+            if (broadcast) {
+               request = invokeCommandOnAll(command, collector, deliverOrder, timeout, TimeUnit.MILLISECONDS);
+            } else {
+               request = invokeCommand(targets, command, collector, deliverOrder, timeout, TimeUnit.MILLISECONDS);
+            }
+         }
       }
+      return request;
    }
 
-   private void addSiteMasterPicker() {
-      RELAY2 relay2 = channel.getProtocolStack().findProtocol(RELAY2.class);
-      if (relay2 != null) {
-         relay2.siteMasterPicker(new SiteMasterPickerImpl());
+   private void addClusterRequest(AbstractRequest<?> request) {
+      try {
+         clusterRequests.addRequest(request);
+         request.onNewView(clusterView.getMembersSet());
+      } catch (Throwable t) {
+         // Removes the request and the scheduled task, if necessary
+         request.cancel(true);
+         throw t;
       }
    }
 
-   @Override
-   public Set<String> getSitesView() {
-      RELAY2 relay = channel.getProtocolStack().findProtocol(RELAY2.class);
-      RouteStatusListener listener = relay != null ?
-            relay.getRouteStatusListener() : null;
-      return (listener instanceof Supplier) ?
-            ((Supplier<Set<String>>) listener).get() : null;
+   public void sendToAll(ReplicableCommand command, DeliverOrder deliverOrder)
```

Comment:
I'm thinking of adding it to the `Transport` interface, as it's more discoverable than `sendToMany(null)`. WDYT?